src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -363,13 +364,60 @@ public void run(final RunNotifier notifier) {
             statement.evaluate();
         } catch (AssumptionViolatedException e) {
             testNotifier.addFailedAssumption(e);
-        } catch (StoppedByUserException e) {
-            throw e;
         } catch (Throwable e) {
-            testNotifier.addFailure(e);
+            processExecutionFailures(testNotifier, gatherAllFailures(e));
         }
     }
 
+    /**
+     * Goes through all execution failures trying to find and re-throw 
+     * {@link StoppedByUserException} while adding all other errors to
+     * test notifier.
+     * 
+     * @param testNotifier notifier to add failures
+     * @param failures list of execution failures
+     */
+    private void processExecutionFailures(EachTestNotifier testNotifier,
+            List<Throwable> failures) {
+        StoppedByUserException stoppedByUserException = null;
+        
+        for (Throwable failure : failures) {
+            if (failure instanceof StoppedByUserException) {
+                stoppedByUserException = (StoppedByUserException) failure;
+            } else {
+                testNotifier.addFailure(failure);
```

Comment:
This is a behavior change. You are sending to the notifier each of the failures that was contained in the `MultipleFailure` exception, when before it was notified once.

It sounds the problem is that a `StoppedByUserException` wrapped in a `MultipleFailureException` doesn't cause `run(RunNotifier)` to exit. Couldn't you check if any of the exceptions inside of the `MultipleFailureException` is a `StoppedByUserException`, and if so, throw that `StoppedByUserException` after sending the `MultipleFailureException` to the notifier?

Also, where is the `MultiFailureException` being created that contains the `StoppedByUserException`? It's possible that code is in error.